src/System.Windows.Forms.Primitives/src/Interop/WinFormsComWrappers.FileOpenDialogWrapper.cs

```diff
@@ -0,0 +1,525 @@
+            {
+                HRESULT result;
+                IntPtr psi_local;
+                if (psi == null)
+                {
+                    psi_local = IntPtr.Zero;
+                }
+                else
+                {
+                    var pUnk_local = WinFormsComWrappers.Instance.GetOrCreateComInterfaceForObject(psi, CreateComInterfaceFlags.None);
+                    var local_psi_IID = IID.IShellItem;
+                    result = (HRESULT)Marshal.QueryInterface(pUnk_local, ref local_psi_IID, out psi_local);
+                    Marshal.Release(pUnk_local);
+                    if (result.Failed())
+                    {
+                        Marshal.ThrowExceptionForHR((int)result);
+                    }
+                }
+
+                result = ((delegate* unmanaged<IntPtr, IntPtr, HRESULT>)(*(*(void***)_wrappedInstance + 12)))
+                    (_wrappedInstance, psi_local);
+                if (result.Failed())
+                {
+                    Marshal.ThrowExceptionForHR((int)result);
+                }
+            }
+
+            void Shell32.IFileOpenDialog.GetFolder(out Shell32.IShellItem ppsi)
+            {
+                HRESULT result;
+                IntPtr ppsi_local;
+                result = ((delegate* unmanaged<IntPtr, IntPtr*, HRESULT>)(*(*(void***)_wrappedInstance + 13)))
+                    (_wrappedInstance, &ppsi_local);
+                if (result.Failed())
+                {
+                    Marshal.ThrowExceptionForHR((int)result);
+                }
+
+                ppsi = ppsi_local == IntPtr.Zero ? null! : (Shell32.IShellItem)WinFormsComWrappers.Instance.GetOrCreateObjectForComInstance(ppsi_local, CreateObjectFlags.Unwrap);
+            }
+
+            HRESULT Shell32.IFileOpenDialog.GetCurrentSelection(out Shell32.IShellItem ppsi)
+            {
+                IntPtr ppsi_local;
+                HRESULT retVal;
+                retVal = ((delegate* unmanaged<IntPtr, IntPtr*, HRESULT>)(*(*(void***)_wrappedInstance + 14)))
+                    (_wrappedInstance, &ppsi_local);
+                ppsi = ppsi_local == IntPtr.Zero ? null! : (Shell32.IShellItem)WinFormsComWrappers.Instance.GetOrCreateObjectForComInstance(ppsi_local, CreateObjectFlags.Unwrap);
+                return retVal;
+            }
+
+            void Shell32.IFileOpenDialog.SetFileName(string pszName)
+            {
+                HRESULT result;
+                fixed (char* pszName_local = pszName)
+                {
+                    result = ((delegate* unmanaged<IntPtr, char*, HRESULT>)(*(*(void***)_wrappedInstance + 15)))
+                        (_wrappedInstance, pszName_local);
+                }
+
+                if (result.Failed())
+                {
+                    Marshal.ThrowExceptionForHR((int)result);
+                }
+            }
+
+            void Shell32.IFileOpenDialog.GetFileName(out string pszName)
```

Comment:
I think this should be 
```suggestion
            void Shell32.IFileOpenDialog.GetFileName(out string? pszName)
```